Fix volume button icon

nswarup14 opened the pull request:
**Issue**
1. The Volume button on the top of the activity failed to load the 'mute' icon on clicking.

**Changes**
1. `activity.py`

**Features**
On clicking the volume button now, the 'mute icon is loaded' and the sound is disabled as well.

nswarup14 committed:
Fix volume button icon

quozl commented:
Thanks.  Does it work for you?  When I try the activity at 22f6848, I get _IOError: [Errno 13] Permission denied score.pkl_ and the PyGame canvas is not drawn.  Only widgets that work are the activity menu, mute button or the stop button.

nswarup14 committed:
Save the score file in the instance directory of the activity

* Earlier the game would not start due to a file permission denied error. The cause was
trying to create a score file in a /usr/.. directory. I have replaced the path of the score file
with that of the activity root instance directory, which has sufficient permissions. Now the activity starts
without any issue.

nswarup14 commented:
@quozl @chimosky
I have fixed the game start issue by changing the file path where the file is created. Kindly review and test.

Tested on Sugar desktop v0.114 and Ubuntu 18.04
The game starts with no trouble.
Thanks!

rhl-bthr commented on main.py:
```diff
@@ -290,16 +291,17 @@ def make(self):
-        if os.path.exists("score.pkl")==False:
-            open('score.pkl','w+')
+        score_path = get_activity_root() + '/instance' + '/score.pkl'
+        if os.path.exists(score_path) is False:
```
`if not os.path.exists(score_path)` would be better

nswarup14 replied:
Fixed it @pro-panda

rhl-bthr commented on main.py:
```diff
@@ -290,16 +291,17 @@ def make(self):
         
-        if os.path.exists("score.pkl")==False:
-            open('score.pkl','w+')
+        score_path = get_activity_root() + '/instance' + '/score.pkl'
```
Please use `os.path.join` to join file paths

nswarup14 replied:
Fixed it @pro-panda

rhl-bthr reviewed (changes requested):
I haven't looked into the issue yet, but `'/instance' + '/score.pkl'` doesn't look like the best place to add the score file

nswarup14 committed:
Minor code corrections

* Use os.path.join instead of regular string concatanation
* Modify if condition, using not

quozl commented:
It doesn't play.  When activity is started, computer fan speed increases.  When I press PLAY, activity locks up, and computer fan speed increases more.  Have you tested this?  I had a brief look.  It does not follow sugargame sample activity design properly.

Looking at your patch, there's a lot of copy and paste coding.  It is an opportunity to factorise score file access into a separate module; why did you not notice this?  :grin:

File path should be _data_ rather than _instance_.  Many activities get this wrong.  See these pages;

* [Low-Level Activity API - Writable Directories](https://wiki.sugarlabs.org/go/Development_Team/Low-level_Activity_API#Writable_Directories),
* [Development Team - Almanac - How do I get the file path where I can write files?](https://wiki.sugarlabs.org/go/Development_Team/Almanac/sugar.activity.activity#How_do_I_get_the_file_path_where_I_can_write_files_programmatically.3F)

nswarup14 commented:
@quozl 
I tested it in two places. Ubuntu 18.04, via the GNOME terminal and Sugar desktop.
Both the places I'm able to run the activity. I shall attach screenshot as well.
![Screenshot from 2019-08-22 07-17-25](https://user-images.githubusercontent.com/34597895/63479741-fb3cde00-c4ac-11e9-958e-87d3cf0c4c5e.png)
![Screenshot from 2019-08-22 07-17-35](https://user-images.githubusercontent.com/34597895/63479742-fb3cde00-c4ac-11e9-963b-1ea97f597d97.png)
![Screenshot from 2019-08-22 07-17-41](https://user-images.githubusercontent.com/34597895/63479744-fbd57480-c4ac-11e9-98fe-a52c5190f8e7.png)

> Looking at your patch, there's a lot of copy and paste coding. It is an opportunity to factorise score file access into a separate module; why did you not notice this?

I noticed actually, but felt it was not that resource intensive, hence it's alright even if I have multiple instances of the score file at different places. I shall refactor.

nswarup14 committed:
Replace instance directory with activity data directory

quozl commented:
Thanks.  I've investigated.  The performance and heating problem is caused by the choice of 60 frames per second animation and that the entire display surface is updated on each frame.  It is more common to update only the changed parts of the frame.  The Sugargame test activity shows how this is done.  As it was before your work began on the activity, I'll open an issue for it.

Tested eb093be.

chimosky commented:
Tested eb093be, comments.

It'll be great if the background of `GAME OVER!` extends to the last part of the word.

quozl commented:
Thanks.  Reviewed.  Will merge and raise feedback as issues.

quozl commented:
@nswarup14, for your interest, this pull request was difficult to complete because the activity was not mature to begin with.  Many problems existed before your work began, and this made our testing difficult.

nswarup14 commented:
@quozl 
Yes, I understand.
I shall work on the other issues that have been opened up now.
